### PR TITLE
Update README.md (v1.4.x)

### PR DESCRIPTION
**Pull request by srochel**
added initial information for MXNet v1.4.0

## Description ##
(Brief description on what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Doc, pr-awaiting-review]

**Comment by srochel** [comment]
@vandanavk - please review one line change and comment or approve.

File: README.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ How to Contribute
 
 What's New
 ----------
+* [Version 1.4.0 Release](to be created - https://github.com/apache/incubator-mxnet/releases/tag/1.4.0) - MXNet 1.4.0 Release.
```
Comment (lebeg) [inline README.md]: Since there is no 1.4.0 tag yet this will be effectively a broken link until the tag will be created. Maybe this part of the change need to be postponed.

**Comment by lebeg** [comment]
And if you don't mind, could you rephrase the title to `Update README.md (v1.4.x)`? It would follow the format `<message> (<branch>)`
